File: be/src/main/java/sg/flow/models/transaction/history/TransactionHistoryDetail.java
package sg.flow.models.transaction.history;

import java.time.LocalDate;
import java.time.LocalTime;


import lombok.Data;
import sg.flow.entities.Account;
import sg.flow.entities.Card;

@Data
public class TransactionHistoryDetail {
    
    private Long id;
    private String transactionReference;

    private Account account;

    private Card card;
    private LocalDate transactionDate;
    private LocalTime transactionTime;
    private String description;
    private Double amount;
    private String transactionType;
    private String transactionStatus;
    private String friendlyDescription;
}

File: be/src/main/kotlin/sg/flow/models/transaction/TransactionHistoryDetail.kt
package sg.flow.models.transaction

import java.time.LocalDate
import java.time.LocalTime
import sg.flow.entities.Account
import sg.flow.models.card.BriefCard

data class TransactionHistoryDetail(
        val id: Long,
        val transactionReference: String? = null,
        val account: Account? = null,
        val card: BriefCard? = null,
        val transactionDate: LocalDate? = null,
        val transactionTime: LocalTime? = null,
        val amount: Double,
        val transactionType: String? = null,
        val description: String,
        val transactionStatus: String? = null,
        val friendlyDescription: String? = null
)
